include/ignition/math/Vector4.hh

```diff
@@ -126,6 +128,56 @@ namespace ignition
         this->data[3] = _w;
       }
 
+      /// \brief Set this vector's components to the maximum of itself and the
+      ///        passed in vector
+      /// \param[in] _v the maximum clamping vector
+      public: void Max(const Vector4<T> &_v)
+      {
+        if (_v[0] > this->data[0])
+          this->data[0] = _v[0];
+        if (_v[1] > this->data[1])
+          this->data[1] = _v[1];
+        if (_v[2] > this->data[2])
+          this->data[2] = _v[2];
+        if (_v[3] > this->data[3])
+          this->data[3] = _v[3];
+      }
+
+      /// \brief Set this vector's components to the minimum of itself and the
+      ///        passed in vector
+      /// \param[in] _v the minimum clamping vector
+      public: void Min(const Vector4<T> &_v)
+      {
+        if (_v[0] < this->data[0])
+          this->data[0] = _v[0];
+        if (_v[1] < this->data[1])
+          this->data[1] = _v[1];
+        if (_v[2] < this->data[2])
+          this->data[2] = _v[2];
+        if (_v[3] < this->data[3])
+          this->data[3] = _v[3];
+      }
+
+      /// \brief Get the maximum value in the vector
+      /// \return the maximum element
+      public: T Max() const
+      {
+        return std::max(std::max(std::max(this->data[0],
```

Comment:
I believe that `std::max_element` would be a bit more concise: https://en.cppreference.com/w/cpp/algorithm/max_element